### PR TITLE
adjust recently merged symbols_collision event and better document it

### PR DESCRIPTION
**Pull request by rafaeldtinoco**
commit edd80f47 (HEAD -> symbols_coll_refactor, rafaeldtinoco/symbols_coll_refactor, main)
Author: Rafael David Tinoco <rafaeldtinoco@gmail.com>
Date:   Sun Feb 19 14:25:27 2023

    symbols_collision: refactor and comment extension
    
    1. containers/path_resolver: return a pointer
    
    2. ebpf/tracee:
       - cleanup derivation table init function
       - move symbols_collision initialization to the correct place
    
    3. events/symbols_loaded:
       - extend init function with code from derivation table init
       - adjust formatting for readability
    
    4. events/symbols_collision:
       - extend init function with code from derivation table init
       - adjust formatting for readability
       - add comments to explain the logic
       - rename variables for meaning clarity
    
    5. events/symbols_collision_test:
       - adjust test to reflect the new init function

commit 0fa6ca53
Author: Rafael David Tinoco <rafaeldtinoco@gmail.com>
Date:   Sun Feb 19 14:24:25 2023

    cmd/flags/filter: nit spelling fix

commit cad2076b
Author: Rafael David Tinoco <rafaeldtinoco@gmail.com>
Date:   Fri Feb 17 13:41:00 2023

    containers: rename PathResolver to ContainerPathResolver

commit 344f1006
Author: Rafael David Tinoco <rafaeldtinoco@gmail.com>
Date:   Thu Feb 16 22:11:18 2023

    tracee: adjust singleton class by adding comments

commit c931a3e8
Author: Rafael David Tinoco <rafaeldtinoco@gmail.com>
Date:   Thu Feb 16 22:14:13 2023

    tracee: advance bucketscache initialization
    
    This is needed for further changes.

### CODE REVIEW

File: pkg/containers/path_resolver.go [hunk 1]
```diff
@@ -10,30 +10,33 @@ import (
...
+// /proc/1/ns, requiring CAP_SYS_PTRACE capability.
+type ContainerPathResolver struct {
 	fs               fs.FS
 	mountNSPIDsCache *bucketscache.BucketsCache
 }
 
-// InitPathResolver create a resolver for paths from within containers.
-func InitPathResolver(mountNSPIDsCache *bucketscache.BucketsCache) PathResolver {
-	return PathResolver{
+// InitContainerPathResolver create a resolver for paths from within containers.
```
Comment (geyslan) [inline pkg/containers/path_resolver.go:21]:
```suggestion
// InitContainerPathResolver creates a resolver for paths from within containers.
```

File: pkg/containers/path_resolver.go [hunk 2]
```diff
@@ -10,30 +10,33 @@ import (
...
 
-// ResolveAbsolutePath resolves an absolute path, which might be inside a container, to an absolute path
-// within the host mount namespace.
-func (cPathRes PathResolver) ResolveAbsolutePath(mountNSAbsolutePath string, mountNS int) (string, error) {
+// GetHostAbsPath translates an absolute path, which might be inside a container, to the
+// correspondent abs path in the host mount namespace.
+func (cPathRes *ContainerPathResolver) GetHostAbsPath(mountNSAbsolutePath string, mountNS int) (
+	string, error,
+) {
 	// path should be absolute, except for e.g memfd_create files
```
Comment (geyslan) [inline pkg/containers/path_resolver.go:34]:
```suggestion
	// path should be absolute, except for, e.g., memfd_create files
```

File: pkg/events/derive/symbols_collision.go [hunk 3]
```diff
@@ -2,231 +2,329 @@ package derive
...
-	processesSOsCache processesLoadedObjectsCache
-	collisionsCache   sharedObjectsCollisionsCache
-	symbolsBlackList  map[string]bool
-	symbolsWhiteList  map[string]bool
-	returnedErrors    map[string]bool
+	soLoader               sharedobjs.DynamicSymbolsLoader
+	loadedObjsPerProcCache loadedObjsPerProcessCache // cache of loaded shared objects per process
+	collisionChecksCache   collisionChecksCache      // cache of symbols collisions
+	symbolsBlacklistMap    map[string]bool           // will delete symbols NOT in this map
+	symbolsWhitelistMap    map[string]bool           // will delete symbols IN this map
```
Comment (geyslan) [inline pkg/events/derive/symbols_collision.go:63]:
```suggestion
	deniedSymbolsMap    map[string]bool           // will delete symbols NOT in this map
	allowedSymbolsMap   map[string]bool           // will delete symbols IN this map
```

And for all related.

**Review by geyslan — APPROVED** [review]
LGTM, just some suggestions.

File: pkg/containers/path_resolver.go (hunk 1 above)
Comment (rafaeldtinoco) [reply pkg/containers/path_resolver.go:21]: ok

File: pkg/containers/path_resolver.go (hunk 2 above)
Comment (rafaeldtinoco) [reply pkg/containers/path_resolver.go:34]: ok

**Pull request merged**